src/main/scala/io/iohk/ethereum/blockchain/sync/RegularSync.scala

```diff
@@ -126,19 +134,27 @@ trait RegularSync {
         //we have same chain prefix
         if (parent.hash == headers.head.parentHash) {
 
-          val currentBranchTotalDifficulty = headersQueue.map(_.number)
-            .map(blockNumber => blockchain.getBlockByNumber(blockNumber))
-            .collect{
-              case Some(b) => b.header.difficulty
-              case None => BigInt(0)
-            }.sum
+          val oldBranch: Seq[Block] = getOldBlocks(headersQueue)
+
+          val oldBlocks: Seq[(BlockBody, BigInt)] = oldBranch
+            .map{b => (b.body, b.header.difficulty)}
```

Comment:
As shown in the code snippet I pasted, this is not needed. In fact it's easier to use `oldBranch` directly.